### PR TITLE
Lock

### PR DESCRIPTION
**Pull request by JulianVennen**

### CODE REVIEW

File: commands/lock.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    await message.react(util.icons.error);
+    return;
+  }
+
+  let channels = await util.channelMentions(message.guild,args);
+  let embed = new Discord.MessageEmbed().setTitle('This channel has been locked!').setDescription(args.join(' ')).setColor(util.color.red);
+  let everyone = message.guild.roles.everyone.id;
+
+  if (channels.length) {
+    let updates = '';
```
Comment (KurtThiemann) [inline commands/lock.js:24]:
```suggestion
    let updates = [];
```

File: commands/lock.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    for(let channel of channels) {
+      channel = message.guild.channels.resolve(channel);
+
+      if (await lock(channel, everyone, embed))
+        updates += `<#${channel.id}>, `;
```
Comment (KurtThiemann) [inline commands/lock.js:29]:
```suggestion
        updates.push(`<#${channel.id}>`);
```

File: commands/lock.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    if (updates.length) {
+      await message.channel.send(`Locked ${updates.substring(0,updates.length - 2)}!`);
```
Comment (KurtThiemann) [inline commands/lock.js:32]:
```suggestion
      await message.channel.send(`Locked ${updates.join(', ')}!`);
```

File: commands/lock.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    else {
+      await message.channel.send(`No channels to lock!`);
+    }
+  }
+  else if (args.length && ['all','global'].includes(args[0].toLowerCase())){
+    args = args.slice(1);
+    let embed = new Discord.MessageEmbed().setTitle('This channel has been locked!').setDescription(args.join(' ')).setColor(util.color.red);
+    channels = bot.guilds.cache.get(message.guild.id).channels.cache;
+    let updates = '';
```
Comment (KurtThiemann) [inline commands/lock.js:42]:
```suggestion
    let updates = [];
```

File: commands/lock.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    for(let [id, channel] of channels) {
+      if (!(channel instanceof Discord.TextChannel)) {
+        continue;
+      }
+
+      if (await lock(channel, everyone, embed))
+        updates += `<#${channel.id}>, `;
```
Comment (KurtThiemann) [inline commands/lock.js:49]:
```suggestion
        updates.push(`<#${channel.id}>`);
```

File: commands/lock.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    if (updates.length) {
+      await message.channel.send(`Locked ${updates.substring(0,updates.length - 2)}!`);
```
Comment (KurtThiemann) [inline commands/lock.js:52]:
```suggestion
      await message.channel.send(`Locked ${updates.join(', ')}!`);
```

File: commands/unlock.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    await message.react(util.icons.error);
+    return;
+  }
+
+  let channels = await util.channelMentions(message.guild,args);
+  let embed = new Discord.MessageEmbed().setTitle('This channel has been unlocked!').setDescription(args.join(' ')).setColor(util.color.green);
+  let everyone = message.guild.roles.everyone.id;
+
+  if (channels.length) {
+    let updates = '';
```
Comment (KurtThiemann) [inline commands/unlock.js:24]:
```suggestion
    let updates = [];
```

File: commands/unlock.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    for(let channel of channels) {
+      channel = message.guild.channels.resolve(channel);
+      if (await unlock(channel, everyone, embed))
+        updates += `<#${channel.id}>, `;
```
Comment (KurtThiemann) [inline commands/unlock.js:28]:
```suggestion
        updates.push(`<#${channel.id}>`);
```

File: commands/unlock.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    }
+    await message.channel.send(`Unlocked ${updates.substring(0,updates.length - 2)}!`);
```
Comment (KurtThiemann) [inline commands/unlock.js:30]:
```suggestion
    await message.channel.send(`Unlocked ${updates.join(', ')}!`);
```

What if no channels were unlocked (updates.length === 0)?

File: commands/unlock.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  }
+  else if (args.length && ['all','global'].includes(args[0].toLowerCase())){
+    args = args.slice(1);
+    let embed = new Discord.MessageEmbed().setTitle('This channel has been unlocked!').setDescription(args.join(' ')).setColor(util.color.green);
+    channels = bot.guilds.cache.get(message.guild.id).channels.cache;
+    let updates = '';
```
Comment (KurtThiemann) [inline commands/unlock.js:36]:
```suggestion
    let updates = [];
```

File: commands/unlock.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    for(let [id, channel] of channels) {
+      if (!(channel instanceof Discord.TextChannel)) {
+        continue;
+      }
+
+      if (await unlock(channel, everyone, embed))
+        updates += `<#${channel.id}>, `;
```
Comment (KurtThiemann) [inline commands/unlock.js:43]:
```suggestion
        updates.push(`<#${channel.id}>`);
```

File: commands/unlock.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+    }
+    if (updates.length) {
+      await message.channel.send(`Unlocked ${updates.substring(0,updates.length - 2)}!`);
```
Comment (KurtThiemann) [inline commands/unlock.js:46]:
```suggestion
      await message.channel.send(`Unlocked ${updates.join(', ')}!`);
```

**Review by KurtThiemann — CHANGES_REQUESTED** [review]

File: commands/lock.js [hunk 13]
```diff
@@ -0,0 +1,105 @@
+const util = require('../lib/util.js');
+const Discord = require('discord.js');
+
+const command = {};
+
+command.description = 'Dissallow users to send messages to a channel';
```
Comment (matthi4s) [inline commands/lock.js:6]:
```suggestion
command.description = 'Disallow users to send messages to one or multiple channels';
```

File: commands/unlock.js (hunk 9 above)
Comment (KurtThiemann) [reply commands/unlock.js]: ?

File: commands/lock.js [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+
+command.execute = async (message, args, database, bot) => {
+  //Permission check
+  if (!util.isMod(message.member) && !message.member.hasPermission('MANAGE_CHANNELS')) {
+    await message.react(util.icons.error);
+    return;
+  }
+
+  let channels = await util.channelMentions(message.guild,args);
+  let embed = new Discord.MessageEmbed().setTitle('This channel has been locked!').setDescription(args.join(' ')).setColor(util.color.red);
```
Comment (matthi4s) [inline commands/lock.js:20]:
```suggestion
  let embed = new Discord.MessageEmbed().setTitle('This channel is locked.').setDescription(args.join(' ')).setColor(util.color.red).setFooter('You are not muted, this channel is locked for everyone. Don't send direct messages to team members or moderators .');
```
(Or a similar footer text, not sure about the exact wording yet.)

**Review by matthi4s — CHANGES_REQUESTED** [review]

File: commands/lock.js [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+    if (updates.length) {
+      await message.channel.send(`Locked ${updates.join(', ')}!`);
+    }
+    else {
+      await message.channel.send(`No channels to lock!`);
+    }
+  }
+  else if (args.length && ['all','global'].includes(args[0].toLowerCase())){
+    args = args.slice(1);
+    let embed = new Discord.MessageEmbed().setTitle('This channel has been locked!').setDescription(args.join(' ')).setColor(util.color.red);
```
Comment (matthi4s) [inline commands/lock.js:40]:
```suggestion
```

File: commands/lock.js [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+  for(const p of ['SEND_MESSAGES', 'ADD_REACTIONS']) {
+    if (permissions.has(p)) {
+      if (channel.permissionOverwrites.get(everyone)) {
+        config.lock[p] = channel.permissionOverwrites.get(everyone).allow.has(p) ? true : null;
+      }
+      else {
+        config.lock[p] = null;
+      }
+      let o = {};
+      o[p] = false;
```
Comment (matthi4s) [inline commands/lock.js:89]: Don't call variables `o` and `p`.

File: commands/lock.js [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+
+command.execute = async (message, args, database, bot) => {
+  //Permission check
+  if (!util.isMod(message.member) && !message.member.hasPermission('MANAGE_CHANNELS')) {
+    await message.react(util.icons.error);
+    return;
+  }
+
+  let channels = await util.channelMentions(message.guild,args);
+  let embed = new Discord.MessageEmbed().setTitle('This channel is locked.').setDescription(args.join(' ')).setColor(util.color.red).setFooter('You are not muted, this channel is locked for everyone. Don't send direct messages to team members or moderators .');
```
Comment (matthi4s) [inline commands/lock.js:20]:
```suggestion
  let embed = new Discord.MessageEmbed().setTitle('This channel is locked.').setDescription(args.join(' ')).setColor(util.color.red).setFooter('You are not muted, this channel is locked for everyone. Don\'t send direct messages to team members or moderators .');
```

**Review by matthi4s — APPROVED** [review]

File: commands/unlock.js (hunk 9 above)
Comment (JulianVennen) [reply commands/unlock.js]: hatte den kommentar übersehen habs gefixt

**Review by KurtThiemann — APPROVED** [review]

**Pull request merged**